src/credentials/CHIPCert.cpp

```diff
@@ -746,6 +757,30 @@ CHIP_ERROR ChipDN::GetCertChipId(uint64_t & chipId) const
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR ChipDN::GetCertFabricId(uint64_t & fabricId) const
+{
+    uint8_t rdnCount = RDNCount();
+
+    fabricId = 0;
+
+    for (uint8_t i = 0; i < rdnCount; i++)
+    {
+        switch (rdn[i].mAttrOID)
+        {
+        case kOID_AttributeType_ChipFabricId:
+            // Ensure only one FabricID RDN present, since start value is 0, which is reserved and never seen.
+            VerifyOrReturnError(fabricId == 0, CHIP_ERROR_WRONG_CERT_TYPE);
+
+            fabricId = rdn[i].mChipVal;
+            break;
+        default:
+            break;
+        }
+    }
+
+    return CHIP_NO_ERROR;
```

Comment:
if none of the OIDs are `kOID_AttributeType_ChipFabricId` is a fabricId of 0 a correct return value? if so please document it here, otherwise please return an error if no fabric ID can be found.

Above you seem to rely that `fabricId 0` is invalid.

Reply:
Updated to return an error if no fabric ID is found.